src/app/features/wallets/wallets.page.ts

```diff
@@ -85,6 +96,73 @@ export class WalletsPage {
     this.shouldHideDepositButton = this.platform.is('ios');
   }
 
+  ionViewDidEnter() {
+    this.processIframeEvents();
+  }
+
+  private processIframeEvents() {
+    fromEvent(window, 'message')
+      .pipe(
+        tap(event => {
+          const postMessageEvent = event as MessageEvent;
+          switch (postMessageEvent.data) {
+            case 'iframe-on-load':
+              this.iframeLoaded$.next(true);
+              break;
+            case 'iframeBackButtonClicked':
+              this.navController.pop();
+              break;
+            case 'iframe-buy-num-button-clicked':
+              this.navigateToBuyNumPage();
+              break;
+            case 'iframe-copy-to-clipboard-asset-wallet':
+              this.copyToClipboardAssetWallet();
+              break;
+            case 'iframe-copy-to-clipboard-integrity-wallet':
+              this.copyToClipboardIntegrityWallet();
+              break;
+            case 'iframe-copy-to-clipboard-private-key':
+              this.copyToClipboardPrivateKey();
+              break;
+            default:
+              break;
+          }
+        }),
+        untilDestroyed(this)
+      )
+      .subscribe();
+  }
+
+  private copyToClipboardAssetWallet() {
+    this.assetWalletAddr$
+      .pipe(
+        first(),
+        concatMap(assetWalletAddr => this.copyToClipboard(assetWalletAddr)),
+        untilDestroyed(this)
+      )
+      .subscribe();
+  }
+
+  private copyToClipboardIntegrityWallet() {
+    this.publicKey$
+      .pipe(
+        first(),
+        concatMap(publicKey => this.copyToClipboard(publicKey)),
+        untilDestroyed(this)
+      )
+      .subscribe();
+  }
+
+  private copyToClipboardPrivateKey() {
+    this.privateKey$
+      .pipe(
+        first(),
+        concatMap(privateKey => this.copyToClipboard(privateKey)),
+        untilDestroyed(this)
+      )
+      .subscribe();
+  }
+
   // eslint-disable-next-line class-methods-use-this
   onBuyNumBtnClicked() {
     Browser.open({
```

Comment:
**ChatGPT response for `git diff 96daee1c..5a35c2c3 "*wallets.page.ts"`**

Overall, the code appears to be following best practices and is well organized. The use of behavior subjects, the extraction of iframe URL, and the addition of iframe event handling are good improvements to make the code more maintainable and scalable.